File: solutions/lua/armstrong-numbers/1/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
	local sum, exp = 0, math.floor(math.log10(number) + 1)

	local i = number
	while i ~= 0 do
		sum, i = sum + ((i % 10) ^ exp), i // 10
	end
	return sum == number
end

return ArmstrongNumbers

File: solutions/zig/armstrong-numbers/1/armstrong_numbers.zig
const math = @import("std").math;

pub fn isArmstrongNumber(num: u128) bool {
    if (num == 0) return true;

    const num_digits = 1 + math.log10_int(num);
    var sum: u128 = 0;

    var i = num;
    while (i != 0 and sum <= num) : (i /= 10) {
        sum += math.pow(u128, i % 10, num_digits);
    }
    return sum == num;
}
